sdk/src/types/block/core/wrapper.rs

```diff
@@ -128,6 +128,30 @@ impl BlockWrapper {
         &self.signature
     }
 
+    /// Returns the work score of a [`BlockWrapper`].
+    pub fn workscore(&self) -> u32 {
+        let workscore_structure = self.protocol_parameters().work_score_structure;
+        let workscore_bytes = workscore_structure.data_kilobyte * self.packed_len() as u32;
+        let mut workscore_kilobytes = workscore_bytes / 1024;
+
+        workscore_kilobytes += self.workscore_header(workscore_structure);
+        workscore_kilobytes += self.block.workscore(workscore_structure);
+        workscore_kilobytes += self.workscore_signature(workscore_structure);
+        workscore_kilobytes
+    }
+
+    #[inline(always)]
+    fn workscore_header(&self, _workscore_structure: WorkScoreStructure) -> u32 {
+        // Header has 0 work score.
+        0
+    }
```

Comment:
No point in having workscore_header() and adding it here if it's 0?

Reply:
we don't know if it stays 0, and should it change we already have prepared a place to change it. Since it's always inlined it should be fine.

Reply:
If it can change it should be a field in the WorkScoreStructure, no?

Reply:
could it be that maybe we just return some constant > 0 at some point? :thinking:

Reply:
Removed it. Once the BlockHeader PR is merged you still of the opinion that it's unnecessary? Same for a `ValidationBlock`. I feel like despite those having 0 work score it's cleaner to add the method for consistency. We do a lot of things for consistency already that wouldn't be strictly needed, so :shrug:

Reply:
I guess the work for it is handled by `block: description: Block accounts for work done to process a block in the node software.` and don't see the point in adding something for consistency, when there is no point in having it, just confusing